src/Vision/VNRecognizedPointsObservation.cs

```diff
@@ -0,0 +1,65 @@
+//
+// VNRecognizedPointsObservation.cs
+//
+// Authors:
+//	Alex Soto  <alexsoto@microsoft.com>
+//
+// Copyright (c) Microsoft Corporation.
+//
+
+using System;
+using Foundation;
+using ObjCRuntime;
+
+namespace Vision {
+	public partial class VNRecognizedPointsObservation {
+
+		public T [] GetAvailableKeys<T> () where T : Enum
+		{
+			var type = typeof (T);
+			if (!(type == typeof (VNHumanBodyPoseObservationJointName) || type == typeof (VNHumanHandPoseObservationJointName)))
+				throw new InvalidOperationException ($"Only '{nameof (VNHumanBodyPoseObservationJointName)}' and '{nameof (VNHumanHandPoseObservationJointName)}' are supported.");
+
+			NSString[] keys = AvailableKeys;
+			if (keys == null)
+				return null;
+
+			var count = keys.Length;
+			var availableKeys = new T [count];
```

Comment:
You're creating this array twice now, since `Array.ConvertAll` also creates the array.

Reply:
hahaha OMG what a shame 🤦  you are completely right, I should not do this late night heh